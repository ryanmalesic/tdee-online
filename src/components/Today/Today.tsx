import React from 'react';

import { Log } from '../../types';

export interface TodayProps {
  log: Log;
}

const Today: React.FC<TodayProps> = (props) => {
  const { log } = props;

  return (
    <div className="box">
      <h1 className="title is-2">Today</h1>
      <h3 className="subtitle is-4">Your log for today</h3>

      <div className="container has-text-centered">
        <div className="level">
          {log === undefined ? (
            <div className="level-item has-text-centered">
              <p className="subtitle is-3">No log for today</p>
            </div>
          ) : (
            <>
              <div className="level-item has-text-centered">
                <p className="subtitle is-3">
                  <strong className="has-text-primary">{log.weight}</strong> lbs
                </p>
              </div>

              <div className="level-item has-text-centered">
                <p className="subtitle is-3">
                  <strong className="has-text-primary">{log.caloricIntake}</strong> kCals
                </p>
              </div>
            </>
          )}
        </div>
      </div>
    </div>
  );
};

export default Today;
