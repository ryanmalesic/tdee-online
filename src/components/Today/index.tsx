import Today from './Today';

export { Today as default };
