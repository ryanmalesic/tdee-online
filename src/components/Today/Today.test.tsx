import React from 'react';
import renderer from 'react-test-renderer';

import Today from './Today';

it('renders Today unchanged', () => {
  const tree = renderer
    .create(<Today log={{ date: '01-01-1980', weight: 170, caloricIntake: 3500 }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
