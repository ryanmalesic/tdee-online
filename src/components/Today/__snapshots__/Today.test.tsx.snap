// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Today unchanged 1`] = `
<div
  className="box"
>
  <h1
    className="title is-2"
  >
    Today
  </h1>
  <h3
    className="subtitle is-4"
  >
    Your log for today
  </h3>
  <div
    className="container has-text-centered"
  >
    <div
      className="level"
    >
      <div
        className="level-item has-text-centered"
      >
        <p
          className="subtitle is-3"
        >
          <strong
            className="has-text-primary"
          >
            170
          </strong>
           lbs
        </p>
      </div>
      <div
        className="level-item has-text-centered"
      >
        <p
          className="subtitle is-3"
        >
          <strong
            className="has-text-primary"
          >
            3500
          </strong>
           kCals
        </p>
      </div>
    </div>
  </div>
</div>
`;
